### PR TITLE
Add API support for transaction metadata on WriteRels and DeleteRels

### PR DESCRIPTION
**Pull request by josephschorr**

### CODE REVIEW

File: internal/services/v1/relationships.go [hunk 1]
```diff
@@ -42,6 +43,8 @@ var writeUpdateCounter = promauto.NewHistogramVec(prometheus.HistogramOpts{
 	Buckets:   []float64{0, 1, 2, 5, 10, 15, 25, 50, 100, 250, 500, 1000},
 }, []string{"kind"})
 
+const MaximumTransactionMetadataSize = 65000 // bytes. Limited by the BLOB size used in MySQL driver
```
Comment (tstirrat15) [inline internal/services/v1/relationships.go:46]: Is it worth making this datastore-specific? Or will we burn that bridge when we get to it?

File: internal/services/v1/relationships.go [hunk 2]
```diff
@@ -367,7 +374,28 @@ func (ps *permissionServer) WriteRelationships(ctx context.Context, req *v1.Writ
...
+		return nil
+	}
+
+	b, err := metadata.MarshalJSON()
+	if err != nil {
+		return err
+	}
+
+	if len(b) > MaximumTransactionMetadataSize {
+		return NewTransactionMetadataTooLargeErr(len(b), MaximumTransactionMetadataSize)
```
Comment (tstirrat15) [inline internal/services/v1/relationships.go:388]: Are we guaranteed that the client encoding will match the JSON encoding? i.e. will the size that that the client sees on their machine match what we're validating against here?

**Review by tstirrat15 — DISMISSED** [review]
See comments, otherwise LGTM

File: internal/services/v1/relationships.go (hunk 2 above)
Comment (josephschorr) [reply internal/services/v1/relationships.go:388]: No; this is a rough estimate but since the drivers store JSON too, it should be sufficient

**Review by tstirrat15 — DISMISSED** [review]

**Review by tstirrat15 — APPROVED** [review]
Reapproving after rebas

**Pull request merged**